### DISCUSSION TITLE
how to generate ProtobufCIntRange

### DISCUSSION THREAD

**Discussion by snikeguo** [protobuf-c/protobuf-c#600] [Q&A]
**how to generate ProtobufCIntRange ?What does ProtobufCIntRange mean？**
i am developing a small rpc  (Embedded MCU).
My idea ：
C# IDL File ->complie (use csscript) ->**reflect** to get all fields, so I don't need to write the parser code. and I intend to support uint8_t /uint8_t array type，
and generate serialization code for protobuf-c.
like this:
//idl file:
```
struct student
{
    string name;
    int age;
   ...
}
interface StudentServiceGroup
{
    student  GetStudent(string name);
}
```
 It will generate code：
client:

struct student
{
    char *name;
    int age;
   ...
}
student_t__pack(.....){............}
ProtobufCFieldDescriptor student_t__field_descriptors[]=........;
student  GetStudent (string name)
{
   ###  ** Serialize string using protobuf-c**
   send(buf) ;
   wait;
   if(ok){ Deserialize;  return};
}

**Comment by snikeguo** [comment]
![image](https://user-images.githubusercontent.com/20830435/77148805-3a76f000-6acb-11ea-9d72-1e96bf70da0e.png)

**Comment by lipnitsk** [comment]
`ProtobufCIntRange` is used internally in the `ProtobufCMessageDescriptor` to optimize message processing. Unless you are constructing ProtobufCMessageDescriptors by hand you should never have to use `ProtobufCIntRange`.
